custom_components/ble_monitor/config_flow.py

```diff
@@ -0,0 +1,157 @@
+"""Config flow for BLE Monitor."""
+import logging
+import voluptuous as vol
+
+from homeassistant.core import callback
+from homeassistant.helpers import config_validation as cv
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_DEVICES,
+    CONF_DISCOVERY,
+    CONF_MAC,
+    CONF_NAME,
+    CONF_TEMPERATURE_UNIT,
+)
+
+from .const import (
+    DEFAULT_ROUNDING,
+    DEFAULT_DECIMALS,
+    DEFAULT_PERIOD,
+    DEFAULT_LOG_SPIKES,
+    DEFAULT_USE_MEDIAN,
+    DEFAULT_ACTIVE_SCAN,
+    DEFAULT_HCI_INTERFACE,
+    DEFAULT_BATT_ENTITIES,
+    DEFAULT_REPORT_UNKNOWN,
+    DEFAULT_DISCOVERY,
+    DEFAULT_RESTORE_STATE,
+    CONF_ROUNDING,
+    CONF_DECIMALS,
+    CONF_PERIOD,
+    CONF_LOG_SPIKES,
+    CONF_USE_MEDIAN,
+    CONF_ACTIVE_SCAN,
+    CONF_HCI_INTERFACE,
+    CONF_BATT_ENTITIES,
+    CONF_REPORT_UNKNOWN,
+    CONF_RESTORE_STATE,
+    CONF_ENCRYPTION_KEY,
+    DOMAIN,
+    MAC_REGEX,
+    AES128KEY_REGEX,
+)
+
+DEVICE_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_MAC): cv.matches_regex(MAC_REGEX),
+        vol.Optional(CONF_NAME): cv.string,
+        vol.Optional(CONF_ENCRYPTION_KEY): cv.matches_regex(AES128KEY_REGEX),
+        vol.Optional(CONF_TEMPERATURE_UNIT): cv.temperature_unit,
+    }
+)
+
+DOMAIN_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_ROUNDING, default=DEFAULT_ROUNDING): cv.boolean,
+        vol.Optional(CONF_DECIMALS, default=DEFAULT_DECIMALS): cv.positive_int,
+        vol.Optional(CONF_PERIOD, default=DEFAULT_PERIOD): cv.positive_int,
+        vol.Optional(CONF_LOG_SPIKES, default=DEFAULT_LOG_SPIKES): cv.boolean,
+        vol.Optional(CONF_USE_MEDIAN, default=DEFAULT_USE_MEDIAN): cv.boolean,
+        vol.Optional(CONF_ACTIVE_SCAN, default=DEFAULT_ACTIVE_SCAN): cv.boolean,
+        # vol.Optional(
+        #     CONF_HCI_INTERFACE, default=[DEFAULT_HCI_INTERFACE]
```

Comment:
I think if a list doesn't work, you can use:

```
vol.Optional(CONF_HCI_INTERFACE, default=DEFAULT_HCI_INTERFACE): cv.positive_int,
```

This will allow only one HCI interface to be setup, but that will be sufficient for most people. It at least gives users an option

Reply:
> Unfortunately, I got

That's why WIP ;)
I probably didn't test all the cases.

Do you have something configured in yaml ?

Reply:
> I think if a list doesn't work, you can use

Yep. Good point. Will do.
I'll also look for device list, but I have little hope

Reply:
I tun 0.118.2. I removed config, restarted and added via UI. Then removed it, added again, but somewhere, it got messed up. Now I have the above error after each restart. Setup failed for ble_monitor: No setup function defined.